src/components/atoms/AddDatatoken.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { DDO } from '@oceanprotocol/lib'
+import Button from '../atoms/Button'
+import { addDatatokenToWallet } from '../../utils/web3'
+import { useWeb3 } from '../../providers/Web3'
+import { getProviderInfo, IProviderInfo } from 'web3modal'
+
+export default function AddToken({ ddo }: { ddo: DDO }): ReactElement {
+  const { web3Provider } = useWeb3()
+  const [providerInfo, setProviderInfo] = useState<IProviderInfo>()
+
+  useEffect(() => {
+    if (!web3Provider) return
+    const providerInfo = getProviderInfo(web3Provider)
+    setProviderInfo(providerInfo)
+  }, [web3Provider])
+
+  return (
+    <div>
```

Comment:
is this wrapping `div` required? Cause when adding component another `div` is wrapped around it. So this component could return the button or `null` only. The goal should always be to keep the DOM free of markup which is not used, so in this case all users without MetaMask would still get that wrapping `div`, and then another one further down when component is used

Reply:
Yeah that makes sense, I'll change this. Thank you for the feedback

Reply:
This is done, it needed some element to wrap it so I used a fragment